src/bot_events.py

```diff
@@ -0,0 +1,92 @@
+from types import SimpleNamespace
+import inspect
+
+_callbacks = { 
+    # even though the hook.Add function will automatically key:value and put it in here if it doesn't exist,
+    # be nice to other developers and make sure you put it in here so they know it exists
+    # thanks
+    
+    # bot hooks
+    "on_bot_init": {},
+
+    "on_create_game": {},
+    "on_end_game": {},
+
+    "on_run_pindle": {},
+    "on_run_shenk": {},
+    "on_run_trav": {},
+    "on_run_nihlathak": {},
+    "on_run_arcane": {},
+    "on_run_diablo": {},
+    "on_run_end": {},
+
+    "on_pause": {},
+    "on_resume": {},
+
+    "on_item_keep": {},
+    "on_run": {},
+    "on_death": {},
+    "on_chicken": {},
+    "on_merc_death": {},
+    "on_failed_run": {},
+
+    "on_self_health_update": {},
+    "on_merc_health_update": {},
+
+    "bot_loop": {},
+
+
+    # websockets / website hooks
+    "recv_website_request": {},
+}
+
+
+"""
+    This is a bit unorthodox, but the way this works is you do
+
+    hook.Call("on_pickup_item", item_name=item, location=location, ocr_text=ocr_text) you would put this line where the bot picks up an item
```

Comment:
You could consider defining *this* interface alongside the name of a callback, possibly alongside type info, and basically validating in `hook.Call` that all the right parameters are passed in or raising a valueError, to make both debugging and using these callbacks easier